fitsio/src/hdu.rs

```diff
@@ -1062,11 +1070,99 @@ hduinfo_into_impl!(i8);
 hduinfo_into_impl!(i32);
 hduinfo_into_impl!(i64);
 
+/// Iterator for the keys of an HDU header.
+pub struct CardIter {
```

Comment:
I wonder if it would be better to update `read_key` to support either a name or index, and support returning the `Card` type so we can return the key as well. Then the iterator would boil down to something like

```rust
// TODO: proper types
struct CardIter {
    idx: usize,
    num_cards: usize,
    fits_file: FitsFile,
    hdu: FitsHdu,
}

impl Iterator for CardIter {
    type Item = Card;

    fn next(&mut self) -> Option<Self::Item> {
        if self.idx >= self.num_cards {
            return None;
        }

        // Generic return type means we can implement `ReadsKey` for `Card` which returns the name, value and (optional) comment
        let card: Card = self.hdu.read_key(&mut self.fits_file, self.idx).unwrap();  // TODO: error handling
        Some(card)
    }
}
```

Reply:
Sounds right, I am new to the both FITS and the `fitsio` API and missed that this was a possibility.